report_common_nrq/report/common_template.xml

```diff
@@ -474,7 +491,21 @@
                                     />
                                 </td>
                                 <td class="text-right">
-                                    <span t-field="l.price_unit" />
+                                    <t t-set="price_unit" t-value="l.price_unit" />
+                                    <t
+                                        t-if="round(price_unit) != price_unit
+                                        or
+                                        not doc.currency_id.rounding >= 1"
+                                    >
+                                        <span t-esc="'{:,.2f}'.format(price_unit)" />
+                                    </t>
+                                    <t
+                                        t-if="round(price_unit) == price_unit
+                                        and
+                                        doc.currency_id.rounding >= 1"
+                                    >
+                                        <span t-esc="'{:,.0f}'.format(price_unit)" />
+                                    </t>
```

Comment:
Any reason why you are not calling the template here?

Reply:
Sorry, nothing to reason.